rapidpro/src/main/java/org/motechproject/rapidpro/webservice/RapidProHttpClient.java

```diff
@@ -0,0 +1,210 @@
+
+    private static final String PROTOCOL = "https://";
+    private static final String DOMAIN_NAME = "app.rapidpro.io/api/";
+    private static final String URL = PROTOCOL + DOMAIN_NAME;
+    private static final String AUTHORIZATION_HEADER = "Authorization";
+    private static final String TOKEN = "Token ";
+
+    private static final String ERROR_URI = "Error building URI for request";
+    private static final String UNSUPPORTED_MEDIA = "Unsupported Media Type: ";
+    private static final String REDIRECTION = "Redirection: ";
+    private static final String CLIENT_ERROR = "Client Error: ";
+    private static final String SERVER_ERROR = "Server Error: ";
+    private static final String EXECUTING_REQUEST = "Executing request: ";
+    private static final String UNABLE_TO_EXECUTE = "Unable to execute request";
+    private static final String ERROR_RESPONSE_ENTITY = "Error retreiving response entity.";
+    private static final String NO_SETTINGS = "No settings";
+    private static final String NO_API_KEY = "No API key";
+    private static final String NO_VERSION = "No API version";
+
+    private static final int STATUS_ERROR_300 = 300;
+    private static final int STATUS_ERROR_400 = 400;
+    private static final int STATUS_ERROR_500 = 500;
+
+    private HttpClient client;
+    private SettingsService settingsService;
+
+    @SuppressWarnings("SpringJavaAutowiringInspection")
+    @Autowired
+    public RapidProHttpClient(HttpClientBuilderFactory httpClientBuilderFactory, SettingsService settingsService) {
+        this.client = httpClientBuilderFactory.newBuilder().build();
+        this.settingsService = settingsService;
+    }
+
+    /**
+     * Executes an HTTP GET Request
+     *
+     * @param endpoint            The API endpoint
+     * @param responseMediaFormat The desired format for the response
+     * @param params              The query parameters
+     * @return An {@link InputStream} containing the response
+     * @throws RapidProClientException
+     */
+    public InputStream executeGet(String endpoint, MediaFormat responseMediaFormat, Map<String, String> params) throws RapidProClientException {
+        URI uri = buildUri(endpoint, responseMediaFormat, params);
+        HttpGet request = new HttpGet(uri);
+        return executeRequestWithResponseContent(request);
+    }
+
+    /**
+     * Executes an HTTP POST Request
+     *
+     * @param endpoint            The API endpoint
+     * @param body                The body of the request
+     * @param contentType         The content type of the request
+     * @param responseMediaFormat The desered format for the response
+     * @return An {@link InputStream} containing the response
+     * @throws RapidProClientException
+     */
+    public InputStream executePost(String endpoint, ByteArrayOutputStream body, MediaFormat contentType, MediaFormat responseMediaFormat) throws RapidProClientException {
```

Comment:
I would prefer this to take a byte array as the param, instead of a specific stream type